api/v1alpha1/clusterconfig_types.go

```diff
@@ -213,6 +216,17 @@ func (CNI) VariableSchema() clusterv1.VariableSchema {
 	}
 }
 
+// NFD tells us to enable or disable the node feature discovery addon.
+type NFD struct{}
+
+func (NFD) VariableSchema() clusterv1.VariableSchema {
+	return clusterv1.VariableSchema{
+		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+			Type: "Object",
```

Comment:
```suggestion
			Type: "object",
```

Reply:
Perhaps we could add some builders for this structure?